packages/playground/data-liberation/src/import/WP_Topological_Sorter.php

```diff
@@ -0,0 +1,461 @@
+<?php
+
+/**
+ * The topological sorter class.
+ *
+ * We create an in-memory index that contains offsets and lengths of items in the WXR.
+ * The indexer will also topologically sort posts so that the order we iterate over posts
+ * ensures we always get parents before their children.
+ */
+class WP_Topological_Sorter {
+
+	/**
+	 * The base name of the table.
+	 */
+	const TABLE_NAME = 'data_liberation_index';
+
+	/**
+	 * The option name for the database version.
+	 */
+	const OPTION_NAME = 'data_liberation_db_version';
+
+	/**
+	 * The current database version, to be used with dbDelta.
+	 */
+	const DB_VERSION = 1;
+
+	// Element types.
+	const ELEMENT_TYPE_POST     = 1;
+	const ELEMENT_TYPE_CATEGORY = 2;
+
+	/**
+	 * Variable for keeping counts of orphaned posts/attachments, it'll also be assigned as temporarly post ID.
+	 * To prevent duplicate post ID, we'll use negative number.
+	 *
+	 * @var int
+	 */
+	protected $orphan_post_counter = 0;
```

Comment:
Do we need to restore this (or any other) value after stopping and resuming the import?